(feat) File viewing

wingio opened the pull request:
- [x] View markdown files
    - [x] Swap to raw view
- [x] View text files
    - [x] Syntax highlighting
    - [x] Select lines
    - [x] Scroll bar
    - [x] Zoom
- [x] View image files
    - [x] Pinch to zoom
    - [x] Download images
- [x] View PDFs
    - [x] Download PDFs

wingio committed:
Add code and markdown file viewing

wingio committed:
Add image file viewing and share button

wingio committed:
Add error ui to file viewer

wingio committed:
Improve code viewer

- Dynamically size the line number instead of using constant value
- Make all lines the same width to prevent jumping when scrolling
- Optimize more values

wingio committed:
feat: Add draggable scrollbar

oSumAtrIX commented:
A huge QoL feature that the official mobile GitHub app lacks, which is a deal breaker for me at least, is the ability to pinch to zoom in or out of the code. You are limited to view the code, line wrapped or not, at a certain font size, but sometimes, you may want to look at a part of the file more closely or see a big overview of the file. Web browsers allow to zoom into sites, including the file viewer of GitHub which is effectively what Materii may want to support natively.

wingio commented:
Oh yeah someone had requested that on discord, guess i forgot to add it to the todo list

wingio commented:
> A search button for files would also be nice! 

Like being able to search within a file?

wingio committed:
Add ability to view PDF files

wingio commented:
> > > A search button for files would also be nice!
> > 
> > Like being able to search within a file?
> 
> like being able to search for a file.

unfortunately githubs api doesn't seem to support that, best i could do is search only within a directory

wingio committed:
feat: Add gestures to the code viewer

- Double tap to show/hide ui
- Long press to start selection
- Single press while in a selection to select more
- Pinch to zoom (min 70%, max 200%)

wingio committed:
Add toolbar options for some file types

wingio committed:
Add ability to view raw markdown files
